lib/mix/lib/mix/dep.ex

```diff
@@ -368,10 +368,17 @@ defmodule Mix.Dep do
       "\n  Ensure they match or specify one of the above in your deps and set \"override: true\""
   end
 
-  def format_status(%Mix.Dep{app: app, status: {:overridden, other}} = dep) do
-    "the dependency #{app} in #{Path.relative_to_cwd(dep.from)} is overriding a child dependency:\n" <>
-      "#{dep_status(dep)}#{dep_status(other)}" <>
-      "\n  Ensure they match or specify one of the above in your deps and set \"override: true\""
+  def format_status(%Mix.Dep{app: app, opts: opts, status: {:overridden, %Mix.Dep{opts: other_opts} = other}} = dep) do
+    problem = "the dependency #{app} in #{Path.relative_to_cwd(dep.from)} is overriding a child dependency:\n" <>
+    "#{dep_status(dep)}#{dep_status(other)}"
+
+    resolution = case Keyword.get(opts, :in_umbrella) || Keyword.get(other_opts, :in_umbrella) do
```

Comment:
You need to check that one of them is from_umbrella and the other is in_umbrella.

Also, I think this error happens when they *diverge* not when they override, so it should be in another branch of format_status.

Finally, please submit a test too. :) Thank you!